python/ray/train/mosaic/mosaic_trainer.py

```diff
@@ -0,0 +1,168 @@
+import inspect
+import os
+from typing import TYPE_CHECKING, Callable, Dict, Optional
+
+from composer.trainer import Trainer
+
+from ray.air import session
+from ray.train.constants import (
+    EVALUATION_DATASET_KEY,
+    TRAIN_DATASET_KEY,
+)
+from ray.air.checkpoint import Checkpoint
+from ray.air.config import DatasetConfig, RunConfig, ScalingConfig
+from ray.train.torch import TorchConfig, TorchTrainer
+from ray.train.trainer import GenDataset
+from ray.util import PublicAPI
+
+if TYPE_CHECKING:
+    from ray.data.preprocessor import Preprocessor
+
+
+@PublicAPI(stability="alpha")
+class MosaicTrainer(TorchTrainer):
```

Comment:
Can we also add this to the Ray Train API ref: https://github.com/ray-project/ray/blob/master/doc/source/train/api.rst

Reply:
added!